salt/_modules/metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,212 @@
+
+    Args:
+        name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Returns:
+        True: if the suspend state has been updated.
+    """
+    return _set_cronjob_suspend(name, namespace, suspend=True, **kwargs)
+
+
+def activate_cronjob(name, namespace, **kwargs):
+    """Activate a CronJob.
+
+    Args:
+        name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Returns:
+        True: if the suspend state has been updated.
+    """
+    return _set_cronjob_suspend(name, namespace, suspend=False, **kwargs)
+
+
+def get_jobs(cronjob_name, namespace, **kwargs):
+    """Get the Jobs created by a CronJob.
+
+    Args:
+        cronjob_name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Returns:
+        list: list of Jobs created by the CronJob.
+    """
+    cronjob_uid = get_cronjob(cronjob_name, namespace, **kwargs)["metadata"]["uid"]
+
+    # Get all Jobs in the namespace
+    jobs = __salt__["metalk8s_kubernetes.list_objects"](
+        kind="Job", apiVersion="batch/v1", namespace=namespace, **kwargs
+    )
+
+    if not jobs:
+        return []
+
+    # Filter Jobs created by the CronJob
+    filtered_jobs = []
+
+    for job in jobs:
+        owner_refs = job["metadata"].get("ownerReferences", [])
+        for owner_ref in owner_refs:
+            if owner_ref.get("uid") == cronjob_uid:
+                filtered_jobs.append(job)
+                break
+
+    return filtered_jobs
+
+
+def stop_jobs(cronjob_name, namespace, wait=False, timeout_seconds=60, **kwargs):
+    """Stop all Jobs created by a CronJob.
+
+    Args:
+        cronjob_name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+        wait (bool, optional): wait for the Jobs to be deleted. Defaults to False.
+        timeout_seconds (int, optional): timeout in seconds to wait for the Jobs to be deleted. Defaults to 60.
+
+    Raises:
+        CommandExecutionError: if the wait timeout is exceeded.
+
+    Returns:
+        list: list of Jobs that have been deleted.
+    """
+    suspend_cronjob(cronjob_name, namespace, **kwargs)
+
+    jobs = get_jobs(cronjob_name, namespace, **kwargs)
+
+    for job in jobs:
+        __salt__["metalk8s_kubernetes.delete_object"](
+            kind="Job",
+            apiVersion="batch/v1",
+            name=job["metadata"]["name"],
+            namespace=namespace,
+            **kwargs,
+        )
+
+    # Wait for jobs to be deleted
+    # grace period is 30 seconds usually
+    if wait:
+        start_ts = time.time()
+        while time.time() - start_ts < timeout_seconds:
+            waiting_jobs = get_jobs(cronjob_name, namespace, **kwargs)
+            if not waiting_jobs:
+                break
+
+            time.sleep(1)
+        else:
+            raise CommandExecutionError(
+                f"Wait timeout exceeded while deleting the following Jobs {jobs} "
```

Comment:
It's `waiting_jobs` that should be used here to me

```suggestion
                f"Wait timeout exceeded while deleting the following Jobs {waiting_jobs} "
```